addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SourceCodeDisclosureScanRule.java

```diff
@@ -646,7 +646,9 @@ public String getName() {
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         // get the body contents as a String, so we can match against it
         String responsebody = msg.getResponseBody().toString();
-
+        if (msg.getResponseHeader().isCss() || msg.getResponseHeader().isJavaScript()) {
+            return;
+        }
```

Comment:
This can be done before the above statement.

Reply:
Also as mentioned in the ticket this should also check `msg.getRequestHeader().isCss()`

Reply:
I am still not comfortable (or) didn't get my head around on why we need to check "request" header in "response" scan. But you guys know more than me, so will go with your suggestion. @thc202

Reply:
The response scan has the whole message, so one can check both.